src/array.jl

```diff
@@ -379,6 +379,15 @@ const StridedCuVector{T} = StridedCuArray{T,1}
 const StridedCuMatrix{T} = StridedCuArray{T,2}
 const StridedCuVecOrMat{T} = Union{StridedCuVector{T}, StridedCuMatrix{T}}
 
+"""
+    pointer(::CuArray, [index=1]; [type=DeviceMemory])
+
+Get the native address of a CUDA array object, optionally at a given location `index`.
+
+By default, the returned pointer will be usable as if pointing to device memory. This
+ensures that the pointer can be used in device code, regardless of where the array is
+actually stored. To get a pointer that can be used in host code, pass `type=HostMemory`.
```

Comment:
What about `UnifiedMemory`?

Reply:
What I mean is what happens if I pass `type=UnifiedMemory`? Is that invalid?

Reply:
Yeah, that's invalid.

Reply:
Do you think that should be clarified?

Reply:
I'd say yes since a previous CI run of my PR failed due to this :stuck_out_tongue:

Reply:
Alright, I added some details. What do you think?